src/main/java/tars/parsers/Parser.java

```diff
@@ -0,0 +1,30 @@
+package tars.parsers;
+
+import tars.tasks.Task;
+
+import java.time.format.DateTimeFormatter;
+
+/**
+ * An abstract class that represents a parser for interpreting task information.
+ *
+ * <p>The {@code Parser} class provides a base for parsing task-related data
+ * and converting it into {@link Task} objects. It includes a predefined date format
+ * and requires subclasses to implement the {@link #parse(String[])} method
+ * to handle specific parsing logic.</p>
+ */
+public abstract class Parser {
```

Comment:
I like this idea of having different Parsers for the different types of tasks, I might implement it into my own iP as well.